### PR TITLE
Version api

### LINKED ISSUE #23: /version endpoint
The /version endpoint is missing

https://gelato.azurewebsites.net/version

### PR DESCRIPTION
**Pull request by jyotsnaravikumar**
## Type of PR

- [ ] Documentation changes
- [X ] Code changes
- [ ] Test changes
- [ ] CI-CD changes
- [ ] GitHub Template changes

## Purpose of PR
Adds the /version endpoint 

## Issues Closed or Referenced
Closes #23

### CODE REVIEW

File: src/main/java/com/microsoft/azure/helium/config/BuildConfig.java [hunk 1]
```diff
@@ -12,10 +18,26 @@
...
     public String getBuildVersion(){
 
-        String buildName = buildProperties.getName();
-        String buildVersion = buildProperties.getVersion();
-        String buildTime =  String.valueOf(buildProperties.getTime().getEpochSecond());
-        System.out.println("buildName " + buildName +"_"+ "buildVersion "+ buildVersion + "_" + "buildTime "+ buildTime );
-        return buildVersion+"."+buildTime ;
+        Instant buildTime = buildProperties.getTime();
+
+        DateTimeFormatter DATE_TIME_FORMATTER = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss")
```
Comment (bartr) [inline src/main/java/com/microsoft/azure/helium/config/BuildConfig.java]: This doesn't appear to be used. Can it be removed?

File: src/main/java/com/microsoft/azure/helium/config/BuildConfig.java [hunk 2]
```diff
@@ -12,10 +18,26 @@
...
+                .withZone(ZoneId.systemDefault());
+
+        System.out.println(DATE_TIME_FORMATTER.format(buildTime));
+
+        //major.minor.MMdd.hhmm
+        String major = buildProperties.getVersion();
```
Comment (bartr) [inline src/main/java/com/microsoft/azure/helium/config/BuildConfig.java]: Can you combine the two date formats?

File: src/main/java/com/microsoft/azure/helium/config/BuildConfig.java [hunk 3]
```diff
@@ -12,10 +18,26 @@
...
+        DateTimeFormatter DATE_TIME_FORMATTER1 = DateTimeFormatter.ofPattern("MMDD")
+                .withZone(ZoneId.systemDefault());
```
Comment (bartr) [inline src/main/java/com/microsoft/azure/helium/config/BuildConfig.java]: does ZoneId.systemDefault() guarantee UTC?  What if the VM is running in a different TZ?

File: src/main/java/com/microsoft/azure/helium/config/BuildConfig.java [hunk 4]
```diff
@@ -12,10 +18,26 @@
...
+        String MMDD = DATE_TIME_FORMATTER1.format(buildTime);
+
+        DateTimeFormatter DATE_TIME_FORMATTER2 = DateTimeFormatter.ofPattern("hhmm")
+                .withZone(ZoneId.systemDefault());
+        String hhmm = DATE_TIME_FORMATTER2.format(buildTime);
+
+        String versionNo = major + "." + MMDD + "." + hhmm;
```
Comment (bartr) [inline src/main/java/com/microsoft/azure/helium/config/BuildConfig.java]: Can you cache this value in the controller and only compute the first time? It won't change during the app lifespan

File: src/main/java/com/microsoft/azure/helium/config/BuildConfig.java [hunk 5]
```diff
@@ -12,10 +18,26 @@
...
+        System.out.println(versionNo);
```
Comment (bartr) [inline src/main/java/com/microsoft/azure/helium/config/BuildConfig.java]: are the system.out.println calls going to "pollute" your logs? Are they necessary?

File: src/main/java/com/microsoft/azure/helium/config/BuildConfig.java (hunk 3 above)
Comment (jyotsnaravikumar) [reply src/main/java/com/microsoft/azure/helium/config/BuildConfig.java]: Instant is an UTC hence I had used SystemDefault, however added with ZoneId.of("UTC") to be explicit as 
        DateTimeFormatter DATE_TIME_FORMATTER1 = DateTimeFormatter.ofPattern("MMDD").withZone( ZoneId.of("UTC"));

File: src/main/java/com/microsoft/azure/helium/config/BuildConfig.java (hunk 4 above)
Comment (jyotsnaravikumar) [reply src/main/java/com/microsoft/azure/helium/config/BuildConfig.java]: Done!

File: src/main/java/com/microsoft/azure/helium/config/BuildConfig.java (hunk 5 above)
Comment (jyotsnaravikumar) [reply src/main/java/com/microsoft/azure/helium/config/BuildConfig.java]: Cleaned!

File: src/main/java/com/microsoft/azure/helium/config/BuildConfig.java (hunk 2 above)
Comment (jyotsnaravikumar) [reply src/main/java/com/microsoft/azure/helium/config/BuildConfig.java]: DateFormatter cannot combine multiple formats in it's parsing , hence this is the best way we can do.
https://docs.oracle.com/javase/8/docs/api/java/time/format/DateTimeFormatter.html

**Pull request merged**